packages/ui-components/src/widgets/Grid/index.tsx

```diff
@@ -2,59 +2,59 @@ import * as React from 'react';
 import {
   compose,
   pure,
+  defaultProps,
   withPropsOnChange,
-  createEagerFactory,
+  createEagerElement,
   setDisplayName,
 } from 'recompose';
 import * as cx from 'classnames';
 import { memoize } from 'lodash';
 
 const styles = require('./styles.css');
 
-const Column = createEagerFactory(
-  ({ className, children, columnClass, columnStyle }: any) => (
-    <div
-      className={cx(styles.column, className, columnClass)}
-      style={columnStyle}
-    >
-      {children}
-    </div>
-  ),
-);
+export type ColumnElement = React.SFC<any> | React.ComponentClass<any> | string;
 
 const getClassName = memoize(columns =>
-  columns.split('|').map(value => styles[`column-${value}`]),
+  columns.split('|').map(value => styles[`column-${value}`])
 );
 
 export const Grid: any = compose(
   pure,
   setDisplayName('Grid'),
+  defaultProps({ columnElement: 'div' }),
   withPropsOnChange(
     ['columns', 'children'],
-    ({ columns, children }: GridType) => {
+    ({ columns, children, columnElement }: GridType) => {
       const classNames = getClassName(columns);
       return {
         children: React.Children.map(
           children,
           (child: any, index: number) =>
             child &&
-            Column({
-              key: child.key,
-              children: child,
-              className: classNames[index] || classNames[0],
-              columnClass: child.props.columnClass,
-              columnStyle: child.props.columnStyle,
-            }),
+            createEagerElement(
```

Comment:
instead of factory (coz we don't know what kind of element to create beforehand)